app/lib/common/widgets/loading_overlay_screen.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:app/common/style/style.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+/// A factory for a loading overlay screen. Credit https://www.greycastle.se/loading-overlay-in-flutter/
+class LoadingOverlayScreen {
+  BuildContext _context;
+
+  LoadingOverlayScreen._create(this._context);
+
+  factory LoadingOverlayScreen.of(BuildContext context) {
+    return LoadingOverlayScreen._create(context);
+  }
+
+  void hide() {
+    Navigator.of(_context, rootNavigator: true).pop();
+  }
+
+  void show() {
+    showDialog(
+        context: _context,
+        barrierDismissible: false,
+        child: WillPopScope(
```

Comment:
Since we have two navigators, we have to be careful not to allow the user to back out of the loading screen and mess up our history. The absence of events when attempting to click or interact with the underlying screens means `barrierDismissible` works. We must use it alongside the `WillPopScrope` to prevent navigation interaction at the same time.